### ISSUE TITLE
unbound-plus: option to return NXDOMAIN instead of 0.0.0.0

### ISSUE THREAD

**Issue by bradleeedmondson** [opnsense/plugins#1842]
[x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
[x] I have searched the existing issues and I'm convinced that mine is new.
[x] When the request is meant for an existing plugin, I've added its name to the title.

**Problem**
The DNS blacklisting in BIND returns NXDOMAIN, while in Unbound Plus it returns 0.0.0.0. In the case of Unbound Plus, my end-users' browsers then try to make a TCP connection to the network address and have to wait for that to time out before giving up and moving on. Some pages will actually wait on this timeout before moving on and finally rendering/revealing everything else in the page all at once (i.e. most of the DOM is hidden until then). In my environment, this seems to cause pageloads to take on Unbound Plus than with BIND + DNSBL.

**Desired Solution/Feature**
I would like to be able to instruct Unbound Plus to return NXDOMAIN rather than a bogus IP address for entries on its DNS blacklist. Personally I think I would prefer that as the default (or even only) behavior, but it may be more elegant to have an option in the Unbound Plus configuration page as to whether the DNSBL rewrites will be to bogus addresses or to NXDOMAIN.

**Options/Alternatives**
- Change plugin behavior to only serve NXDOMAIN
- Add global option to return 0.0.0.0 IP (current behavior) address or NXDOMAIN (global to Unbound, i.e. not specific per-blocked-domain)
  - Could also add global option Option to specify a local IP address to use as honeypot or traffic sink. Some folks on the forum seemed to want to do that,[1] though it's not in scope for my own use case. (NB: I also posted on that forum thread, about the primary NXDOMAIN question, but there was at least one other person looking for this redirect, a la PiHole)
- Leave as is (folks like me can adapt the DNSBL import script locally if we're able and so inclined)

**Additional context**
For what it's worth, this made a big enough difference in my perceived page load time that I reverted from the architecturally simpler Unbound Plus DNSBL to my previous configuration of [incoming] --> [Unbound (w/ dhcp names)] --> BIND (w/ DNSBL serving NXDOMAIN). If this option were added to the Unbound Plus plugin, I could stop routing queries through a BIND instance on the loopback address.

[1] https://forum.opnsense.org/index.php?topic=14116.msg77577#msg77577

**Comment by schreibubi** [comment]
#1901 changes the behavior to NXDOMAIN

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.
